Algorithmia/CLI.py

```diff
@@ -0,0 +1,233 @@
+import Algorithmia
+import os
+from Algorithmia.algo_response import AlgoResponse
+import json, re, requests
+
+class CLI():
+	def __init__(self):
+		self.client = Algorithmia.client()
+		# algo auth
+	def auth(self, apikey):
+
+		#setenv = "export ALGORITHMIA_API_KEY=" + "'" + APIkey + "'"
+		#print(setenv)
+
+		#store api key in local config file and read from it each time a client needs to be created
+		key = open(os.environ['HOME']+"/.algorithmia_api_key","w")
+		key.write(apikey)
+		key.close()
+
+		print("apikey is:")
+		key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+		print(key.read())
+		key.close()
+
+	# algo run <algo> <args..>    run the the spesified algo
+	def runalgo(self, name, inputs, client):
+		algo_name = name
+
+		algo_input = inputs[0]
+
+		algo = client.algo(algo_name)
+		
+		result = None
+		'''
+		time = 300
+		if("--timeout" in inputs):
+			#find timeout value
+
+
+			time = 300
+
+		algo.set_options(timeout=time, stdout=("--debug" in inputs), output=)
+		'''
+
+
+		#handle input type flags
+		if(len(inputs) >= 1):
+			if(inputs[0] == "-d" or inputs[0] == "--data"):
```

Comment:
There's a great python package called '[argparse](https://pypi.org/project/argparse/)' that can really assist you with making the CLI a bit easier to read. I believe it's bundled into the python standard library for anything > 3.4, which means we'd need to pip install for python2 support. Let me know if I can help you with this

Reply:
Is there a compelling reason to support python2 at all for the CLI? [Python2 ended support _finally_ on Jan 1 this year](https://www.python.org/doc/sunset-python-2/).

Reply:
Just because it's officially ended doesn't mean half of data scientist's laptops / desktops don't use python2 as the default, including mac. I think for tools that are a bit further into the on-boarding experience can drop python2 compatibility, but as this could be an initial touch point for prospects I'm wary.

Reply:
I won't try to speak to data scientists and their need to support python2 client (though, I do think we should document exactly what versions of python are supported, and presumably the client should be tested accordingly - but none of that has anything to do with this PR)

That said, it seems to me that the client library could support older python (so that python2 code can use it) without expecting the CLI to. Even if python2 is the default on a system, you can have an `algo` script that starts with `#!/bin/env python3`. Again, otherwise, you end up needing to test that the CLI works for multiple versions.